slack-clone-compose-sample/src/main/kotlin/io/getstream/slack/compose/ui/channels/ChannelItem.kt

```diff
@@ -0,0 +1,197 @@
+package io.getstream.slack.compose.ui.channels
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.wrapContentWidth
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Icon
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.compose.ui.common.avatar.UserAvatar
+import io.getstream.chat.android.compose.ui.theme.ChatTheme
+import io.getstream.chat.android.compose.ui.util.getDisplayName
+import io.getstream.slack.compose.R
+import io.getstream.slack.compose.ui.channels.components.OnlineIndicator
+import io.getstream.slack.compose.ui.channels.components.UnreadCountBadge
+import io.getstream.slack.compose.ui.util.getOtherUser
+
+/**
+ * Component that represents a one-to-one channel item.
+ *
+ * One-to-one channel is a distinct channel with only two members.
+ *
+ * @param channel The channel to display.
+ * @param onChannelClick Handler for a single tap on an item.
+ * @param modifier Modifier for styling.
+ */
+@Composable
+fun DirectOneToOneChatItem(
+    channel: Channel,
+    onChannelClick: (Channel) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    Row(
+        modifier = modifier
+            .clip(shape = RoundedCornerShape(6.dp))
+            .clickable(onClick = { onChannelClick(channel) })
+            .padding(horizontal = 8.dp)
+            .fillMaxSize(),
+        verticalAlignment = Alignment.CenterVertically,
+    ) {
+        val user = channel.getOtherUser()!!
```

Comment:
Is there any way to avoid this? Maybe passing the "other user" as a parameter once we realize this is a 1:1 chat